include/extractor/edge_based_node.hpp

```diff
@@ -15,43 +15,48 @@ namespace osrm
 namespace extractor
 {
 
+struct RoadSegment
```

Comment:
We could use the space savings and use a little bit of that inline the coordinates in the struct. That would be 16 bytes more, but would solve the following ugliness:

- Speed: It will probably be much faster, no double-indirection
- No need for a separate std::vector<std::pair<NodeID, NodeID>> on construction
- No cyclic dependency on `datafacade`
- No dependency on the coordinates array